### PR TITLE
 config-manager: improve systemd unit listing

### PR DESCRIPTION
**Pull request by fmuyassarov**
Don't rely on ListUnitsByPatternsContext method as it might be too new
and missing in some of the distros like Amazon Linux 2. Instead
filter out active units and then loop over those to check if either
containerd or cri-o units exist.

### CODE REVIEW

File: cmd/config-manager/main.go [hunk 1]
```diff
@@ -180,22 +180,30 @@ func detectRuntime() (string, *dbus.Conn, error) {
...
+	units, err := conn.ListUnitsFilteredContext(context.Background(), []string{"active"})
+
+	expectedUnits := make([]dbus.UnitStatus, 0)
+	for _, unit := range units {
+		if unit.Name == "containerd.service" || unit.Name == "crio.service" {
+			expectedUnits = append(expectedUnits, unit)
+		}
+	}
+
 	if err != nil {
```
Comment (kad) [inline cmd/config-manager/main.go:192]: probably error handling would be better before `make()`

File: cmd/config-manager/main.go [hunk 2]
```diff
@@ -180,22 +180,30 @@ func detectRuntime() (string, *dbus.Conn, error) {
 	// (either containerd or CRI-O).If more than one container runtime systemd unit is found
 	// to be in an active state, the process fails.
 	log.Infof("looking for active runtime units on D-Bus...")
-	units, err := conn.ListUnitsByPatternsContext(context.Background(), []string{"active"}, []string{containerdUnit, crioUnit})
+	units, err := conn.ListUnitsFilteredContext(context.Background(), []string{"active"})
+
+	expectedUnits := make([]dbus.UnitStatus, 0)
+	for _, unit := range units {
+		if unit.Name == "containerd.service" || unit.Name == "crio.service" {
```
Comment (klihub) [inline cmd/config-manager/main.go:187]: nitpick: There are `containerdUnit` and `crioUnit` consts defined for these, so probably those should be used here instead.

File: cmd/config-manager/main.go (hunk 1 above)
Comment (klihub) [reply cmd/config-manager/main.go:192]: +1

**Review by klihub — CHANGES_REQUESTED** [review]
Small nit, otherwise LGTM.

File: cmd/config-manager/main.go (hunk 1 above)
Comment (fmuyassarov) [reply cmd/config-manager/main.go]: Ops, I accidentally moved already existing error handling down. But I have now fixed it.

File: cmd/config-manager/main.go (hunk 2 above)
Comment (fmuyassarov) [reply cmd/config-manager/main.go]: Good catch. Fixed.

**Review by klihub — APPROVED** [review]
LGTM.

**Review by kad — APPROVED** [review]
Lgtm

**Pull request merged**